src/components/ChatLog.js

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import ChatEntry from './ChatEntry';
+import './ChatLog.css';
+import PropTypes from 'prop-types';
+
+const ChatLog = ( props ) => {
+
+    const [entries, setEntries] = useState(props.entries);
+
+    const setLiked = (id, value) => {
+        let numLikes = false;
+        entries.forEach(element => {
+            if(element.id === id){
+            element.liked = value;
+            } 
+            if(element.liked) numLikes++;
+        });
+        setEntries([...entries]); // Force re-render https://stackoverflow.com/a/56266640
+        props.setTotalLikes(numLikes);
+    }
```

Comment:
Here's a slight refactoring of your code that avoids modifying the existing state. This will pass the test in App in its original form:

```javascript
    const setLiked = (id, value) => {
        let numLikes = 0;
        let newEntries = entries.map(element => {
            if (element.id === id) {
                return { ...element, liked: value };
            }
            return element;
        });
        newEntries.forEach(element => { 
            if(element.liked) numLikes++;
        });
        setEntries(newEntries);
        props.setTotalLikes(numLikes);
    }
```

Reply:
The important thing that's different here is: the element that is being modified is *itself* copied. As opposed to JUST the `entries` array being copied, which doesn't copy each of the elements.